### ISSUE TITLE
Stacked ensembler: use same CV data splitter as the rest of automl

### ISSUE THREAD

**Issue by dsherry** [alteryx/evalml#1592]
**Problem**
Currently, the stacked ensembler has its own setup for CV. `IterativeAlgorithm` calls [the `_make_stacked_ensembler` util](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/utils.py#L221), but doesn't currently thread through the data splitter from automl search.

[The data splitter set up by default in the stacked ensembler](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/components/ensemble/stacked_ensemble_base.py#L53) doesn't set `shuffle=True`, which [could lead to poor performance](https://github.com/alteryx/evalml/issues/1259) if the input dataset has an ordering. It also won't have the same settings for other parameters like `n_folds`, which isn't ideal.

Also, this difference [is preventing us from supporting sklearn 0.24.0](https://github.com/alteryx/evalml/pull/1591). Fixing this issue should allow us to support that version.

**Fix**
Let's have automl pass its data splitter through `IterativeAlgorithm` down into the stacked ensembler.

**Comment by dsherry** [comment]
@angela97lin does my explanation make sense / was there a reason we chose not to do this when we were setting up stacking? :)

**Comment by angela97lin** [comment]
@dsherry I think your explanation makes sense! IIRC when we were setting up stacking and were trying to make it more performant / make stacking run faster, we wanted to default to something that didn't have too many folds--hence the `self._default_cv(n_splits=3, random_state=random_state)` line where we take the default specified by scikit-learn, and hardcoded `n_splits` to 3.

**Comment by angela97lin** [comment]
Dug into this a bit more, and tried to weave the data split method used by AutoML to the stacked ensemble component. However, I ran into this issue (after addressing the API updates necessary to get our `TrainingValidationSplit` class to work):
```

estimator = WrappedSKClassifier(pipeline=LogisticRegressionBinaryPipeline(parameters={'Imputer':{'categorical_impute_strategy': 'm...Logistic Regression Classifier':{'penalty': 'l2', 'C': 1.0, 'n_jobs': -1, 'multi_class': 'auto', 'solver': 'lbfgs'},}))
X =            0         1         2         3         4
0   0.965469  0.041236  0.028701  0.659165  0.213375
1   0.043831...978  0.079577
48  0.376344  0.920154  0.314640  0.180086  0.197598
49  0.682661  0.046529  0.400513  0.412513  0.751464
y = array([1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0,
       1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0,
       1, 0, 1, 0, 1, 0])

    @_deprecate_positional_args
    def cross_val_predict(estimator, X, y=None, *, groups=None, cv=None,
                          n_jobs=None, verbose=0, fit_params=None,
                          pre_dispatch='2*n_jobs', method='predict'):
        """Generate cross-validated estimates for each input data point

        The data is split according to the cv parameter. Each sample belongs
        to exactly one test set, and its prediction is computed with an
        estimator fitted on the corresponding training set.

        Passing these predictions into an evaluation metric may not be a valid
        way to measure generalization performance. Results can differ from
        :func:`cross_validate` and :func:`cross_val_score` unless all tests sets
        have equal size and the metric decomposes over samples.

        Read more in the :ref:`User Guide <cross_validation>`.

        Parameters
        ----------
        estimator : estimator object implementing 'fit' and 'predict'
            The object to use to fit the data.

        X : array-like of shape (n_samples, n_features)
            The data to fit. Can be, for example a list, or an array at least 2d.

        y : array-like of shape (n_samples,) or (n_samples, n_outputs), \
                default=None
            The target variable to try to predict in the case of
            supervised learning.

        groups : array-like of shape (n_samples,), default=None
            Group labels for the samples used while splitting the dataset into
            train/test set. Only used in conjunction with a "Group" :term:`cv`
            instance (e.g., :class:`GroupKFold`).

        cv : int, cross-validation generator or an iterable, default=None
            Determines the cross-validation splitting strategy.
            Possible inputs for cv are:

            - None, to use the default 5-fold cross validation,
            - int, to specify the number of folds in a `(Stratified)KFold`,
            - :term:`CV splitter`,
            - An iterable yielding (train, test) splits as arrays of indices.

            For int/None inputs, if the estimator is a classifier and ``y`` is
            either binary or multiclass, :class:`StratifiedKFold` is used. In all
            other cases, :class:`KFold` is used.

            Refer :ref:`User Guide <cross_validation>` for the various
            cross-validation strategies that can be used here.

            .. versionchanged:: 0.22
                ``cv`` default value if None changed from 3-fold to 5-fold.

        n_jobs : int, default=None
            Number of jobs to run in parallel. Training the estimator and
            predicting are parallelized over the cross-validation splits.
            ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
            ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
            for more details.

        verbose : int, default=0
            The verbosity level.

        fit_params : dict, defualt=None
            Parameters to pass to the fit method of the estimator.

        pre_dispatch : int or str, default='2*n_jobs'
            Controls the number of jobs that get dispatched during parallel
            execution. Reducing this number can be useful to avoid an
            explosion of memory consumption when more jobs get dispatched
            than CPUs can process. This parameter can be:

                - None, in which case all the jobs are immediately
                  created and spawned. Use this for lightweight and
                  fast-running jobs, to avoid delays due to on-demand
                  spawning of the jobs

                - An int, giving the exact number of total jobs that are
                  spawned

                - A str, giving an expression as a function of n_jobs,
                  as in '2*n_jobs'

        method : {'predict', 'predict_proba', 'predict_log_proba', \
                  'decision_function'}, default='predict'
            The method to be invoked by `estimator`.

        Returns
        -------
        predictions : ndarray
            This is the result of calling `method`. Shape:

                - When `method` is 'predict' and in special case where `method` is
                  'decision_function' and the target is binary: (n_samples,)
                - When `method` is one of {'predict_proba', 'predict_log_proba',
                  'decision_function'} (unless special case above):
                  (n_samples, n_classes)
                - If `estimator` is :term:`multioutput`, an extra dimension
                  'n_outputs' is added to the end of each shape above.

        See Also
        --------
        cross_val_score : Calculate score for each CV split.
        cross_validate : Calculate one or more scores and timings for each CV
            split.

        Notes
        -----
        In the case that one or more classes are absent in a training portion, a
        default score needs to be assigned to all instances for that class if
        ``method`` produces columns per class, as in {'decision_function',
        'predict_proba', 'predict_log_proba'}.  For ``predict_proba`` this value is
        0.  In order to ensure finite output, we approximate negative infinity by
        the minimum finite float value for the dtype in other cases.

        Examples
        --------
        >>> from sklearn import datasets, linear_model
        >>> from sklearn.model_selection import cross_val_predict
        >>> diabetes = datasets.load_diabetes()
        >>> X = diabetes.data[:150]
        >>> y = diabetes.target[:150]
        >>> lasso = linear_model.Lasso()
        >>> y_pred = cross_val_predict(lasso, X, y, cv=3)
        """
        X, y, groups = indexable(X, y, groups)

        cv = check_cv(cv, y, classifier=is_classifier(estimator))
        splits = list(cv.split(X, y, groups))

        test_indices = np.concatenate([test for _, test in splits])
        if not _check_is_permutation(test_indices, _num_samples(X)):
>           raise ValueError('cross_val_predict only works for partitions')
E           ValueError: cross_val_predict only works for partitions

../venv/lib/python3.7/site-packages/sklearn/model_selection/_validation.py:845: ValueError
```

This is an error thrown when we try to call:
```
clf = StackedEnsembleClassifier(input_pipelines=[logistic_regression_binary_pipeline_class(parameters={})], cv=TrainingValidationSplit())
    clf.fit(X, y)
```

The reason for this is because scikit-learn validates that the cv passed is indeed a cross-validation method; it isn't happy with single splits such as `TrainingValidationSplit` where some of the data will never be in the test data (since there is only one split). 

As such, I think the best plan for now is to do the easy thing to support scikit-learn 0.24 and set the default cv's `shuffle=True`. We can revisit this if we think it's a useful thing to do. Thoughts, @dsherry?

Related: https://stackoverflow.com/questions/41753795/sklearn-timeseriessplit-cross-val-predict-only-works-for-partitions

**Comment by angela97lin** [comment]
#1593 should no longer be blocked by this issue, since what was necessary for 0.24.0 should have been resolved in #1613.
